packages/common/src/projects/HubProjects.ts

```diff
@@ -280,3 +289,68 @@ export async function enrichProjectSearchResult(
 
   return result;
 }
+
+/**
```

Comment:
This is the function to really look at initially. I've left lots of comments and I think I've made some correct assumptions, but pls add comments if I'm wrong. Once we're agreed this is the logic we want, I'll optimize it and hoist out some helper fns.